### DISCUSSION TITLE
Getting Involved

### DISCUSSION THREAD

**Discussion by ldonkers** [keepassxreboot/keepassxc#7253] [Q&A] [answered]
Hello,

I am looking to be able to contribute to this project.  I don't really have a programming background but am interested.  Are there any tips for beginners for getting into to open source?  Maybe start with helping with documentation?

Thanks

**Comment by droidmonkey** [comment, marked as answer]
You can work on our documentation! We could use more documentation of our lesser used features and getting new people setup successfully. I'd also like to incorporate our browser troubleshooting guide directly into the user guide. @varjolintu can point you there. 

Our documentation us written using asciidoctor syntax https://asciidoctor.org/ which is similar to markdown with more control. 

You can see our documentation source here: https://github.com/keepassxreboot/keepassxc/tree/develop/docs

If you want to help with code you'll have to be proficient in C++ and Qt5.

**Reply by ldonkers** [reply]
Thanks for the information.  I will check out the documentation.

**Reply by varjolintu** [reply]
https://github.com/keepassxreboot/keepassxc-browser/wiki/Troubleshooting-guide + other wiki pages definitely need some work. A lot of things there would definitely be helpful for the User Guide.

**Reply by ldonkers** [reply]
I will take a look at asciidoctor site and get that installed.  I am completely new to this.  Once I get that installed and work with it I'll have more specific questions as to how you guys use it.  Do you have any pointers to help me get started?

Lennard

**Reply by droidmonkey** [reply]
Use visual studio code and install the asciidoctor extension so that you can preview your changes directly in the editor. It will also let you build an html version to view in the browser. The preview is generally not perfect, but good enough so you get the idea. There are some idioms of asciidoctor like adding a blank line with + to connect continuous numbered sections. You'll see that in our topics as you review what's there.

**Reply by ldonkers** [reply]
I did install Visual Studio Code and was able to convert some files over.  I think the process is a little to complex and not sure if I can be of much help.
